src/modules/accounting/extra_data.py

```diff
@@ -0,0 +1,147 @@
+import itertools
+from dataclasses import dataclass
+from enum import Enum
+from itertools import islice
+
+from hexbytes import HexBytes
+
+from src.web3py.extentions.lido_validators import NodeOperatorIndex
+from src.web3py.typings import Web3
+
+
+class Lengths:
+    ITEM_INDEX = 3
+    ITEM_TYPE = 2
+    MODULE_ID = 3
+    NODE_OPS_COUNT = 8
+    NODE_OPERATOR_IDS = 8
+    STUCK_AND_EXITED_VALS_COUNT = 16
+
+
+class ItemType(Enum):
+    EXTRA_DATA_TYPE_STUCK_VALIDATORS = 0
+    EXTRA_DATA_TYPE_EXITED_VALIDATORS = 1
+
+
+class FormatList(Enum):
+    # TODO old contracts have 1, new contracts have 0, don't forget to change it
+    EXTRA_DATA_FORMAT_LIST_EMPTY = 1
+    EXTRA_DATA_FORMAT_LIST_NON_EMPTY = 1
+
+
+@dataclass
+class ItemPayload:
+    module_id: bytes
+    node_ops_count: bytes
+    node_operator_ids: bytes
+    vals_counts: bytes
+
+
+@dataclass
+class ExtraDataItem:
+    item_index: bytes
+    item_type: ItemType
+    item_payload: ItemPayload
+
+
+@dataclass
+class ExtraData:
+    extra_data: bytes
+    data_hash: HexBytes
+    format: int
+    items_count: int
+
+
+def chunks(it, size):
+    it = iter(it)
+    return iter(lambda: tuple(islice(it, size)), ())
+
+
+class ExtraDataService:
+    def __init__(self, w3: Web3):
+        self.w3 = w3
+
+    def collect(
+        self,
+        exited_validators: dict[NodeOperatorIndex, int],
+        stuck_validators: dict[NodeOperatorIndex, int],
+        max_items_in_payload_count: int,
+        max_items_count: int,
+    ) -> ExtraData:
+        stuck_payloads = self.build_validators_payloads(stuck_validators, max_items_in_payload_count)
```

Comment:
To be consistent with a new version, please check out the following:

https://github.com/lidofinance/lido-dao/blob/e57517730c3e11a41e9cbc32ce018726722335b7/contracts/0.8.9/sanity_checks/OracleReportSanityChecker.sol#L54..L60